dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/RegexUtils.java

```diff
@@ -26,7 +26,7 @@
  */
 public class RegexUtils {
 
-    private static final Pattern LINUX_USERNAME_PATTERN = Pattern.compile("^[a-zA-Z0-9_].{0,30}");
+    private static final Pattern LINUX_USERNAME_PATTERN = Pattern.compile("[a-zA-Z_][a-zA-Z\\d_-]{0,30}");
```

Comment:
```suggestion
    private static final Pattern LINUX_USERNAME_PATTERN = Pattern.compile("[a-zA-Z\\d_-].{0,30}");
```

Reply:
Could you please add ut for this?

Reply:
I think only '-' matches [a-zA-Z\\d_-].{0,30}, pure numbers are also matches [a-zA-Z\\d_-].{0,30};  such as '-',  123456; but these are not valid linux user names!

Reply:
Looking forward to your reply and suggestions

Reply:
The regex failed with `10000`?

Reply:
regex would not failed with 10000, but 10000 is not a valid linux user name

Reply:
Please fix the UT
```
RegexUtilsTest.testIsValidLinuxUserName
```

Reply:
Could you add me wechat or call me, I want to ask you a question? 15651806509

Reply:
> Could you add me wechat or call me, I want to ask you a question? 15651806509

It is better to keep it public so that others can participate in the review.